src/main/src/main/java/org/geoserver/config/util/XStreamPersister.java

```diff
@@ -2529,4 +2530,27 @@ public String toString(Object obj) {
             return obj.toString();
         }
     }
+
+    /** Converter for WMSLayerInfoImpl class ensure backwards compatibility with old directories */
+    class WMSLayerInfoLegacyConverter extends AbstractCatalogInfoConverter {
```

Comment:
This is the WMSLayerInfo converter, no need to add legacy. Or... next time we add a parameter that needs defaulting, are we going to call it LegacyLegacyConverter?

Reply:
will change name to WMSLayerInfo